src/sortStudents.ts

```diff
@@ -1,16 +1,81 @@
-
 export interface Student {
   // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
   // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getAverageGrade(student: Student) : number {
+  return student.grades.reduce((prev:number, cur: number) => prev + cur, 0)
+    / student.grades.length;
+}
 
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder) : Student[] {
   // write your function
+  const studentsCopy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return studentsCopy.sort(
+        (prev: Student, cur: Student) => {
+          return order === 'asc'
+            ? prev.name.localeCompare(cur.name)
+            : cur.name.localeCompare(prev.name);
+        },
+      );
+
+    case SortType.Surname:
+      return studentsCopy.sort(
+        (prev: Student, cur: Student) => {
+          return order === 'asc'
+            ? prev.surname.localeCompare(cur.surname)
+            : cur.surname.localeCompare(prev.surname);
+        },
+      );
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return studentsCopy.sort(
        (prev: Student, cur: Student) => {
          return order === 'asc'
            ? prev[sortBy]localeCompare(cur[sortBy])
            : cur[sortBy]localeCompare(prev[sortBy]);
        },
      );
```
You can make it shorter that way. At checklist this was mentioned.

Reply:
Agree, thats why you add value to your enums